### ISSUE TITLE
SDAT file flipped horizontally at export

### ISSUE THREAD

**Issue by jfbourdon** [jblindsay/whitebox-tools#37]
When converting a DEP file to SDAT format, the exported raster is flipped horizontally. The problem is only with the Python API as the SDAT export form Whitebox GAT 3.4 doesn't have this problem.
```python
wbt.convert_raster_format('infile.dep', 'outfile.sdat')
```
I tested with the lastest compiled binaries (0.16.0). Input and output exemple files can be downloaded [here](https://transfert.mffp.gouv.qc.ca/?ShareToken=EC92CB8691DFC4F0D769EBDEB6D1C696D7E5845C).

**Comment by manmatej** [comment]
I am experiencing the same behaviour when running tool BreachDepressions. I have too large raster to fit GeoTIFF format, so only possibility for me is SAGA grid format, but output is flipped nort-south. I am running from CMD. When I tested smaller tiles with output of GeoTIFF it worked perfectly. It seems problem is SAGA grid format. 

`whitebox_tools -r=BreachDepressions -v --wd=$SCRATCHDIR --dem=dem.sgrd --output=breached.sgrd --max_length=100 --fill_pits=true --flat_increment=0.001
`
[Pictures here. ](https://photos.app.goo.gl/EN9FTyy3aHjpUgv27)

**Comment by jblindsay** [comment]
I believe that this has now been fixed. If someone who uses SAGA GIS could confirm the fix, I will then be able to close this issue.

**Comment by jfbourdon** [comment]
I just ran the conversion with WhiteboxTools 1.0.2, but the result is still flipped horizontally when displayed in both SAGA (7.3.0) and QGIS (3.8.1).

**Comment by jblindsay** [comment]
@jfbourdon Okay, thanks for reporting. This is a surprising tricky one for me. I guess I don't use Saga files often enough. Anyhow, I'll look further into it and see what I can do. Can I ask, are you reading and writing a SAGA file format or are you reading from something else like a GeoTIFF and outputting to a SDAT? I'm just trying to track down where the issue might be (my encoder or decoder).

**Comment by jfbourdon** [comment]
It must be in the SDAT writer because I get the same flipped raster if the input is GeoTIFF, FLT or DEP.

**Comment by jblindsay** [comment]
Well this was a strange bug because it seemed to only occur when accessing the tool(s) from the Python front-end. I could not replicate it calling ConvertRasterFormat from the WhiteboxTools Runner. I honestly can't imagine why this would be the case since in both cases it is the same back-end code that is running. Nonetheless, I have made some more recent changes to the SAGA encoder/decoder with the latest v1.1.0 and have re-tested this particular issue. I now find that the output SDAT files are in fact upright as expected. I would appreciate a SAGA user confirming this before I close the issue however.

**Comment by jfbourdon** [comment]
I tested with v1.1.0 and I get the same results: when calling whitebox_tools.exe, the SDAT is upright, but when using the Python API it's upside down. I also tested with v0.15.0 where both methods give an upside down SDAT but with v0.16.0, I get the same results as v1.1.0. Could it be a Windows/Linux thing?

**Comment by jblindsay** [comment]
I don't think that it could be an operating system related issue. Honestly, this is one of the strangest bugs that I have ever encountered. Keep in mind, whether you call the WhiteboxTools executable from Python, R, QGIS, ArcGIS, or the WhiteboxTools Runner, it is the same backend compiled binary that is running. It really should affect the output of a tool whatever the front-end is. And I don't understand why it is just the SAGA file that is experiencing this issue.  I mean it must have something to do with SAGA's very odd 'toptobottom' flag in its raster header file. However, why would that change depending on whether the tool is being called from Python or the runner? It just makes so little sense. This one is driving me a bit bonkers. I'm almost at the stage where I'm just willing to cut all SAGA support from WhiteboxTools and call it a day.

Can you please tell me precisely what your workflow is. That is, what tool are you running? What does your Python script look like? What are you using to visualize the output, QGIS or SAGA?

**Comment by jfbourdon** [comment]
Well, I found the problem and it's on my computer. This bug was really corrected since v0.16.0 . Several months ago, I downloaded WhiteboxTools v0.11.0 and put the path in my PATH environment variable. So even if I was using `sys.path.append(WBT)` with `WBT` pointing to a newer WhiteboxTools, my call `from whitebox_tools import WhiteboxTools` was using the older one because it was in front of the newer one in the line.

I'm sorry for the trouble, it was really my bad. Monday I'll check if the issues I reported after this one are still valid.
